ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java

```diff
@@ -139,24 +138,19 @@
   private final KsqlExecutionContext executionContext;
   private final RoutingFilterFactory routingFilterFactory;
   private final RateLimiter rateLimiter;
-  private final Optional<PullQueryExecutorMetrics> pullQueryMetrics;
 
   public PullQueryExecutor(
       final KsqlExecutionContext executionContext,
       final RoutingFilterFactory routingFilterFactory,
       final KsqlConfig ksqlConfig,
-      final String serviceId,
-      final Time time
+      final String serviceId
```

Comment:
Instead of passing the things we want to measure, like the startTime or the requestSize, I figured it's cleaner to pass the pullQueryMetrics. Also, the response cannot be measured from in here

Reply:
I think the idea was that PullQueryExecutorMetrics was encapsulated here, but I agree that you can't do everything that way.  Similarly, we cut corners in measuring time by putting that in here to maintain encapsulation.  Andy felt strongly about this, so I made it that way, but I agree that if we want to measure size, it should be moved out.  I think we should measure time in exactly the same manner since we're already pulling PullQueryExecutorMetrics out.